### DISCUSSION TITLE
Can one run LSPs in the scratch buffer?

### DISCUSSION THREAD

**Discussion by bcspragu** [helix-editor/helix#11916] [Q&A]
The use case I have in mind is adding the [Harper](https://github.com/elijah-potter/harper/) spellcheck + grammar check LSP in my scratch buffer, where I regularly take ad hoc notes. I'm mostly trying to understand if this is supported currently, or if limitations of LSP (e.g requiring actual file paths?) would make this tricky

**Comment by kirawi** [comment]
Yes, LSPs require a URI so it's not possible to run them on scratch buffers.

**Reply by setop** [reply]
Helix could create a temporary file, line in tmp in order for the LSP to work a intended

**Reply by TornaxO7** [reply]
> Helix could create a temporary file, line in tmp in order for the LSP to work a intended

Yes, but this can be easily done by the user as well: `hx /tmp/test.py` or `hx /tmp/test.js` (or what other kind of LSP you'd like to start). It's better, in my opinion, to let the user customize that by themself.
I have this fish aliases for example:
```fish
    alias ea '$EDITOR /tmp/test.s'
    alias ec '$EDITOR /tmp/test.c'
    alias ed '$EDITOR /tmp/test.dot'
    alias eja '$EDITOR /tmp/Test.java'
    alias ejs '$EDITOR /tmp/test.json'
    alias ela '$EDITOR /tmp/test.tex'
    alias elu '$EDITOR /tmp/test.lua'
    alias em '$EDITOR /tmp/test.md'
    alias en '$EDITOR $(tv)'
    alias ep '$EDITOR /tmp/test.py'
    alias et '$EDITOR /tmp/test.txt'
    alias ev '$EDITOR /tmp/test.vim'
```

**Reply by setop** [reply]
That's very true. It is even better as in this case the file has an URI but nothing is written on disk before `:w` is issued.
